### PR TITLE
Bump aks-operator to 1.0.1-rc17

### PR DESCRIPTION
**Pull request by thedadams**
Issue:
https://github.com/rancher/rancher/issues/33446

### CODE REVIEW

**Review by rmweir — APPROVED** [review]
LGTM

**Review by cmurphy — APPROVED** [review]

File: pkg/apis/go.mod [hunk 1]
```diff
@@ -6,13 +6,12 @@ replace k8s.io/client-go => k8s.io/client-go v0.21.0
...
 require (
 	github.com/pkg/errors v0.9.1
-	github.com/rancher/aks-operator v1.0.1-rc14
+	github.com/rancher/aks-operator v1.0.1-rc17
 	github.com/rancher/eks-operator v1.1.1-rc3
 	github.com/rancher/fleet/pkg/apis v0.0.0-20210608014113-99e848822739
 	github.com/rancher/gke-operator v1.1.1-rc7
 	github.com/rancher/norman v0.0.0-20210608202517-59b3523c3133
 	github.com/rancher/rke v1.3.0-rc8
-	github.com/rancher/system-upgrade-controller/pkg/apis v0.0.0-20210424054953-634d28b7def3
```
Comment (cbron) [inline pkg/apis/go.mod:15]: This is unrelated ? SUC code isn't used in aks-operator afaik.

**Comment by cbron** [comment]
Closing in favor of other PR
